src/Image.h

```diff
@@ -37,8 +37,8 @@ using JPEGDecodeL = std::function<uint32_t (uint8_t* const src)>;
 class Image: public Nan::ObjectWrap {
   public:
     char *filename;
-    int width, height;
-    int naturalWidth, naturalHeight;
+    size_t width, height;
+    size_t naturalWidth, naturalHeight;
```

Comment:
I'm trying to find any cases where `int` isn't big enough. The only one I found is reading a jpeg, where it's `unsigned`. GifLib, Cairo and librsvg all use int as far as I can tell. Was this type itself responsible for overflow?

One concern making this `size_t` is that `width` and `height` are passed directly to e.g. `cairo_image_surface_create(format, int width, int height)`.

Reply:
If the image is `32798x32798` it will overflow and allocate way to little bytes since:

<img width="210" alt="Screenshot 2019-03-16 at 10 12 57" src="https://user-images.githubusercontent.com/189580/54473764-1572e180-47d4-11e9-8c0f-0378369a5012.png">

Good point about passing to cairo, I think that we already check for some specific value since cairo only accepts a maxiumum width or height of `2^15 - 1` (I think, ref: https://github.com/Automattic/node-canvas/blob/3d81e0ae45ad205540060d3f58ebad463f9d89a0/src/Image.cc#L610-L613)

Okay, I just realized that the image I'm loading has a larger width & height than that, maybe we should just check that before even setting `naturalWidht` and `naturalHeight` at all 🤔

Reply:
I meant to propose: keep the members as `int` (or `unsigned`) since that's the largest Cairo can take, and only cast to a larger type where you multiply them and risk overflow.